### DISCUSSION TITLE
Communication Error (Carelink Follower)

### DISCUSSION THREAD

**Discussion by paroxysmalism** [NightscoutFoundation/xDrip#3775] [Bug reports] [answered]
"Error communicating with Carelink Server. Please try again later". Carelink seems to be up, however.

**Comment by Navid200** [comment]
Hi @benceszasz, 
Any idea what's going on?
Thanks

**Comment by mattster98** [comment]
It doesn't even seem to be opening the browser interface to do the login.  Not sure if this happened immediately after upgrading to Android 15 or not.

**Comment by benceszasz** [comment]
I have just tested a new login and everything is working fine for me.
Are those affected by this issue using Android 15?
I don't have Android 15 yet.

**Reply by Navid200** [reply]
Thanks

**Comment by paroxysmalism** [comment]
I'm still on Android 14


________________________________

**Reply by benceszasz** [reply]
Which country do you use? What happend, why did you have to login? Normally it is enough to login once and then you can use that login forever except if you don't have internet connection for 1 week, because the  CareLink Follower must be able to renew the login, which is valid for 1 week.

**Reply by paroxysmalism** [reply]
I'm in the US. I tried to go through the recaptcha again as a troubleshooting attempt because I wasn't getting any readings on xdrip+. I thought I may have been logged out somehow.

**Reply by solanix** [reply]
I am currently on Android 13. Everything was working fine up until yesterday. I am in the US as well. I also attempted to log in and receive a popup dialog with the following text:
Communication Error!
Error communicating with CareLink Server! Please try again later!

benceszasz: I will check your page to see if there is a new build. I uninstalled and reinstalled the nightly from the main branch. Nothing appeared the resolve the issue on my end. It could be that the US servers changed the site again.

**Comment by benceszasz** [comment]
@jamorham
Could you change the CarePartner app discovery url?
This is just one line of code in
```
package com.eveningoutpost.dexdrip.cgm.carelinkfollow.auth
public class CareLinkAuthenticator
```
Old discovery url:
```
protected static final String CAREPARTNER_APP_DISCO_URL = "https://clcloud.minimed.com/connect/carepartner/v6/discover/android/3.1";
```
New working discovery url (version changed from v6 to v11 and 3.1 to 3.2):
```
protected static final String CAREPARTNER_APP_DISCO_URL = "https://clcloud.minimed.eu/connect/carepartner/v11/discover/android/3.2";
```

**Reply by Navid200** [reply]
@benceszasz It may be easier if you open a PR and test it.  Then, it will be easier for JamOrHam to review and approve.
I would do it myself if I could test.  But, I don't have the sensor.
Of course it is up to you.

**Reply by benceszasz** [reply]
I was just able to check the change in the official app and do a quick test with a US and EU account, everything was working for me with both accounts, regions. 
This url is a starting point for CareLink login, this json configuration contains the logon and authentication parameters (SSO login configurations, regions, countries, etc.). It is a public URL, you can check it in a browser, both the old and the new (.com and .eu are the same):

https://clcloud.minimed.com/connect/carepartner/v6/discover/android/3.1
https://clcloud.minimed.com/connect/carepartner/v11/discover/android/3.2

If you check these jsons there is a SSOConfiguration parameter for US and EU. In case of US the value of the SSOConfiguration was https://carelink.minimed.com/configs/v1/sso_cp_us_v6.json in the old disco json, which is no longer available. The new value in the new disco json SSOConfiguration for US is https://carelink.minimed.com/configs/v1/sso_cp_us_v5.json which is available. You can check it in a browser.

The problem is that I have to leave now and I will not able to do a PR and a test report neither today nor tomorrow, earliest on Monday evening. Until this is changed no one will be able to use CareLink Follower in the US. Since this is a small change, I have made a quick test and it can be easily tested. I hope this could be an exception.

**Reply by Navid200** [reply]
OK,
I will open the PR and will see if I can find someone to test it.
Thanks

**Reply by benceszasz** [reply]
I have my phone with me, I can send you test accounts. Messenger would be good for you?

**Reply by Navid200** [reply]
Email would be good
xDrip dot Navid at gmail dot com

**Reply by benceszasz** [reply]
email sent from carelinkfollower at gmail dot com

**Reply by Navid200** [reply]
@benceszasz Thanks a lot for your support

**Comment by Navid200** [comment, marked as answer]
Please try the Nightly that was released today: https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.11.17

**Reply by mattster98** [reply]
Working for me!  Thanks!

**Reply by paroxysmalism** [reply]
I'm in! I'm currently in sensor warmup but it all looks good. Thanks!

**Reply by Navid200** [reply]
Great!
The answer was really provided by bencezasz.

**Comment by mitchnic** [comment]
Seems some changes have been made at the Medtronic end and xDrip+ can't login with the same error as above. Getting it on my tablet, as well as multiple reports on the FB group. Seems the Discovery URL might have changed again.

**Reply by m0rt4l1n** [reply]
See #4318
